packages/@lwc/synthetic-shadow/src/shared/node-ownership.ts

```diff
@@ -58,6 +58,8 @@ export function getNodeNearestOwnerKey(node: Node): number | undefined {
         }
         host = parentNodeGetter.call(host) as ShadowedNode | null;
 
+        // Elements slotted from top level light DOM into synthetic shadow
+        // reach the slot tag from the shadow element first
```

Comment:
I didn't understand this comment. Are we short-circuiting here so that we don't need to traverse to the root?